packages/ui/primitives/document-dialog.tsx

```diff
@@ -0,0 +1,56 @@
+'use client';
```

Comment:
We should move this from `/primitives` to something like `/components`. What separates a primitive from a component? I don't know, but this is more specific to a document rather than a generic slap whatever at it.